openquake/risklib/scientific.py

```diff
@@ -131,6 +131,10 @@ def __init__(self, vf_id, imt, imls, mean_loss_ratios, covs=None,
                     pass
                 elif lr > 1:
                     raise ValueError('The meanLRs must be <= 1, got %s' % lr)
+                elif cov == 0:
```

Comment:
We also need to have a check prohibiting negative covs. This applies also for the 'LN' distribution.

For the `cov==0` case for both the 'BT' and 'LN' distributions, there are three subcases that must be handled differently:
1. All covs are zero, in which case we switch to the DegenerateDistribution. The situation in #6105 should be identical to this subcase.
2. `cov==0`, but also `lr==0`: this subcase is handled in the `if` statement just above.
3. `cov==0` with a non-negative `lr`: this is the subcase where we finally raise the `ValueError`.

Reply:
Also related to #4841, but at that time we only implemented the check on the upper bound.

Reply:
Pointed out correctly by Maria Camila, there is also a 4th subcase for the 'BT' distribution:
`cov==0`, but `lr==1`: for this subcase, the beta distribution again becomes a one-point degenerate distribution with a Dirac delta function spike at the right end; the vulnerability function should always return a loss ratio of 1.0 in this subcase.

Reply:
All cases described above now handled, except the last one where `cov=0`, `lr=1` for the 'BT' distribution. This combination will be flagged as invalid. 

To correctly handle such cases, we must use a mixed discrete-continuous probability distribution, defined by a discrete probability mass assigned to `lr=0` and `lr=1`, and a continuous probability density for `lr∈(0, 1)`. Out of the scope of this PR.